### PR TITLE
test: add commands to run all test suite on InstantSearch

### PR DESCRIPTION
**Pull request by yannickcr**
Run the test suite on the different flavors to check that we did not break anything while doing a change on the tests.

The tests are run on SauceLabs on CircleCI and can be run locally or on SauceLabs using the `yarn test:local` and `yarn test:saucelabs` commands.

### CODE REVIEW

File: tests/react-instantsearch.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+set -e
+
+rm -rf react-instantsearch
+git clone git@github.com:algolia/react-instantsearch.git --depth=1
```
Comment (tkrugg) [inline tests/react-instantsearch.sh]: Here I think we can add `--branch` to make it clear which branch we're testing.

File: package.json [hunk 2]
```diff
@@ -25,6 +25,9 @@
   "scripts": {
     "commit": "git-cz",
     "version": "conventional-changelog --preset angular --infile CHANGELOG.md --same-file && git add CHANGELOG.md",
+    "test": "yarn test:local",
+    "test:local": "yarn link && cd tests && ./instantsearch.js.sh && ./react-instantsearch.sh && ./vue-instantsearch.sh",
+    "test:saucelabs": "yarn link && cd tests && (export SAUCELABS=1 && ./instantsearch.js.sh && ./react-instantsearch.sh && ./vue-instantsearch.sh)",
```
Comment (francoischalifour) [inline package.json]: Would this work?

```suggestion
    "test:saucelabs": "SAUCELABS=1 yarn test:local",
```

File: tests/instantsearch.js.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+set -e
+
+rm -rf instantsearch.js
+git clone git@github.com:algolia/instantsearch.js.git --branch master --depth=1
```
Comment (francoischalifour) [inline tests/instantsearch.js.sh]: Should we use the `develop` branch for this repo?

File: package.json (hunk 2 above)
Comment (yannickcr) [reply package.json]: I was convinced it was not working but after testing it seems to work after all :+1:

**Pull request merged**